ch/chproto/server_info.go

```diff
@@ -29,10 +33,14 @@ func (srv *ServerInfo) ReadFrom(rd *Reader) (err error) {
 	if err != nil {
 		return err
 	}
-	if timezone != "UTC" {
-		return fmt.Errorf("ch: ClickHouse server uses timezone=%q, expected UTC", timezone)
+
+	serverTimeZone, err := time.LoadLocation(timezone)
+	if err != nil {
+		return err
 	}
 
+	_, timeZoneOffset = time.Now().In(serverTimeZone).Zone()
```

Comment:
This is racy and won't work if you want to connect to multiple servers that have different timezones.